tests/conftest.py

```diff
@@ -0,0 +1,47 @@
+import lightkube
+import pytest
+from _pytest.config.argparsing import Parser
+
+
+def pytest_addoption(parser: Parser):
+    parser.addoption(
+        "--file",
+        help="Path to bundle file to use as the template for tests.  This must include all charms"
+        "built by this bundle, where the locally built charms will replace those specified. "
+        "This is useful for testing this bundle against different external dependencies. "
+        "e.g. ./releases/1.6/kubeflow-bundle.yaml",
+    )
+
+    parser.addoption(
+        "--channel",
+        help="Kubeflow channels, e.g. latest/stable, 1.6/beta",
+    )
+
+
+@pytest.fixture(scope="module")
+def lightkube_client(ops_test):
+    yield lightkube.Client(
+        namespace=ops_test.model_name,
+    )
+
+
+@pytest.fixture(scope="module")
+def deploy_cmd(request, ops_test):
+    if ops_test.model_name != "kubeflow":
+        raise ValueError("kfp must be deployed to namespace kubeflow")
+
+    bundle_file = request.config.getoption("file", default=None)
+    channel = request.config.getoption("channel", default=None)
+
+    if (bundle_file is None and channel is None) or (bundle_file and channel):
```

Comment:
Just in case `bundle_file` or `channel` arrive here as empty strings (which are also invalid), how about:
```suggestion
    if (not bundle_file and not channel) or (bundle_file and channel):
```